repose-aggregator/commons/commons-utilities/src/main/scala/org/openrepose/commons/utils/servlet/http/HttpServletRequestWrapper.scala

```diff
@@ -271,8 +245,46 @@ class HttpServletRequestWrapper(originalRequest: HttpServletRequest, inputStream
     *
     * @return the parameter map for this request
     */
-  override def getParameterMap: util.Map[String, Array[String]] =
-    parameterMap.map(_.asJava).getOrElse(super.getParameterMap)
+  override def getParameterMap: util.Map[String, Array[String]] = {
+    def retrieveParameterMap: util.Map[String, Array[String]] = {
+      // As per Servlet Spec 3.1 section 3.1.1, form parameters are only parsed under certain conditions.
+      if (getContentLength > 0 && "application/x-www-form-urlencoded".equalsIgnoreCase(getContentType) &&
+        ("POST".equalsIgnoreCase(getMethod) || "PUT".equalsIgnoreCase(getMethod))) {
+        val updatedParameterMap = mutable.Map.empty[String, Array[String]]
+        formParameterMap match {
+          case Some(fpm) =>
+            updatedParameterMap ++= fpm
+          case None =>
+            // This is the only place the input stream has a chance of being replaced or manipulated directly by the wrapper.
+            inputStream synchronized {
+              // As per Servlet Spec 3.1 section 3.1.1, form parameters are only available until the input stream is read.
+              if (status == RequestBodyStatus.Available) {
+                if (!inputStream.markSupported) {
```

Comment:
We decided that we weren't going to reset the input stream ourselves. We were going to consume it if you asked for the parameters like everyone else does. If you want to be able to reset it, the implementer of the wrapper is responsible for wrapping the stream themselves before creating the wrapper.